PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -50,26 +50,48 @@ wstring commafyInt(size_t n)
 	return ss.str();
 }
 
-std::string getFileContent(const wchar_t *file2read)
+std::string getFileContent(const wchar_t* file2read, bool* pbFailed)
 {
+	if (pbFailed) *pbFailed = false; // reset
+
 	if (!doesFileExist(file2read))
+	{
+		if (pbFailed) *pbFailed = true;
 		return "";
+	}
 
-	const size_t blockSize = 1024;
-	char data[blockSize];
-	std::string wholeFileContent = "";
-	FILE *fp = _wfopen(file2read, L"rb");
+	FILE* fp = _wfopen(file2read, L"rb");
 	if (!fp)
+	{
+		if (pbFailed) *pbFailed = true;
 		return "";
-
+	}
+	
+	static constexpr size_t blockSize = 1024 * 4; // 4K is optimal chunk for memory, cache, disk or network
+	char data[blockSize];

```

Comment:
Please use smart pointer instead:
```cpp
auto data = std::make_unique<char[]>(blockSize);
```

Reply:
@donho 
Do you really want changing the current stack memory variable to heap one?
I thought that the 4K buffer isn't enough to consider it, so I left there the original stack var.